File: backend/app/schemas/form_submission.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.form_submission import FormSubmissionStatus


class FormSubmissionRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    form_key: str
    name: str
    phone: str | None
    email: str | None
    property_type_interest: str | None
    message: str | None
    status: FormSubmissionStatus
    assigned_to: str | None
    ip_address: str | None
    submitted_at: datetime


class FormSubmissionUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    status: FormSubmissionStatus | None = None
    assigned_to: str | None = None

File: frontend/types/formSubmission.ts
export type FormSubmissionStatus = "new" | "contacted" | "closed";

export interface FormSubmission {
  id: string;
  form_key: string;
  name: string;
  phone: string | null;
  email: string | null;
  property_type_interest: string | null;
  message: string | null;
  status: FormSubmissionStatus;
  assigned_to: string | null;
  ip_address: string | null;
  submitted_at: string;
}
